packages/serialize/types/index.d.ts

```diff
@@ -7,18 +7,23 @@ import * as CSS from 'csstype'
 export { RegisteredCache, SerializedStyles }
 
 export type CSSProperties = CSS.PropertiesFallback<number | string>
-export type CSSPropertiesWithMultiValues = {
+export type CSSPropertiesWithMultiValues<Props = unknown> = {
   [K in keyof CSSProperties]:
     | CSSProperties[K]
     | ReadonlyArray<Extract<CSSProperties[K], string>>
+    | ((
+        props: Props
+      ) => CSSProperties[K] | ReadonlyArray<Extract<CSSProperties[K], string>>)
```

Comment:
Instead of `Extract<CSSProperties[K], string>` I'd just do this `CSSProperties[K] & string`. The same change could be applied at line 13

Reply:
I wonder why not just `CSSProperties[K] | ReadonlyArray<CSSProperties[K]>`? Some of the `CSSProperties[K]` already includes the `(string & {})` so is it better to rely on CSSProperties[K]?

Reply:
Yeah, I don't really know why this is filtered - but since it is, I'd probably prefer to keep it **here** (in the spirit of keeping this PR focused). I'm open to accepting a PR that would remove this filtering - that can be created now or after we land this PR here. It doesn't matter in what order we'd land them - the other one will have to be synced after the first one lands though.

Reply:
Done!